contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -103,6 +106,23 @@ func (k *k8smScheduler) killTask(taskId string) error {
 	return err
 }
 
+func (k *k8smScheduler) terminateTask(task *podtask.T) error {
+	terminateMessage := struct {
+		ID                         string
+		DeletionTimestamp          *util.Time `json:"deletionTimestamp,omitempty"`
+		DeletionGracePeriodSeconds *int64     `json:"deletionGracePeriodSeconds,omitempty"`
+	}{task.ID, task.Pod.DeletionTimestamp, task.Pod.DeletionGracePeriodSeconds}
+
+	msg, err := json.Marshal(terminateMessage)
+	if err != nil {
+		return err
+	}
+
+	data := fmt.Sprintf("%s:%s", messages.TaskTerminate, string(msg)) //TODO(jdef) use a real message type
+	_, err = k.internal.driver.SendFrameworkMessage(task.Executor.GetExecutorId(), mutil.NewSlaveID(task.Spec.SlaveID), data)
```

Comment:
nevermind, i see that we're passing data through to the executor so that it can tell the kubelet about graceful termination.

framework messages aren't reliable. while this implementation is probably fine for now as a first step, we'll want to file a ticket to add reliability guarantees around the delivery of messages like this.